### ISSUE TITLE
ActionGroup: Missing required property 'resourceGroupName'

### ISSUE THREAD

**Issue by yellowhat** [pulumi/pulumi-azure#887]
Hi,
I am trying to deploy a `azure:monitoring/actionGroup` resource.
But I am getting the following error:
```console
  azure:monitoring:ActionGroup (anf_action_group):
    error: azure:monitoring/actionGroup:ActionGroup resource 'anf_action_group' has a problem: Missing required property 'resourceGroupName'
```

The following is from `pulumi up --logtostderr -v=9 2> out.txt`:
```console
I0510 11:18:38.180002     259 eventsink.go:59] resource registration prepared: ty=azure:monitoring/actionGroup:ActionGroup, name=anf_action_group
I0510 11:18:38.180232     259 eventsink.go:62] eventSink::Debug(<{%reset%}>resource registration prepared: ty=azure:monitoring/actionGroup:ActionGroup, name=anf_action_group<{%reset%}>)
I0510 11:18:38.181715     259 eventsink.go:59] top-level input property translated: location -> location
I0510 11:18:38.181895     259 eventsink.go:62] eventSink::Debug(<{%reset%}>top-level input property translated: location -> location<{%reset%}>)
I0510 11:18:38.182700     259 eventsink.go:59] top-level input property translated: location -> location
I0510 11:18:38.182883     259 eventsink.go:62] eventSink::Debug(<{%reset%}>top-level input property translated: location -> location<{%reset%}>)
I0510 11:18:38.184947     259 source_eval.go:316] handling default provider request for package azure-4.1.0
I0510 11:18:38.185995     259 source_eval.go:316] handling default provider request for package azure-4.1.0
I0510 11:18:38.190573     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: emailAddress={me@email.com}
I0510 11:18:38.186094     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:core/resourceGroup:ResourceGroup,hl_rg)]: location={westeurope}
I0510 11:18:38.191967     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:core/resourceGroup:ResourceGroup,hl_rg)]: name={rg}
I0510 11:18:38.192197     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:core/resourceGroup:ResourceGroup,hl_rg)]: Project={dev}
I0510 11:18:38.192407     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:core/resourceGroup:ResourceGroup,hl_rg)]: tags={map[Project:{dev}]}

I0510 11:18:38.191673     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: name={send2me}
I0510 11:18:38.193304     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: emailReceivers={[{map[emailAddress:{me@email.com} name:{send2me}]}]}
I0510 11:18:38.193445     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: name={anf_email}
I0510 11:18:38.193569     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: resourceGroupName=output<string>{}
I0510 11:18:38.193707     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: shortName={anf_email}
I0510 11:18:38.193828     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: Project={dev}
I0510 11:18:38.193949     259 rpc.go:247] Unmarshaling property for RPC[ResourceMonitor.RegisterResource(azure:monitoring/actionGroup:ActionGroup,anf_action_group)]: tags={map[Project:{dev}]}
```

Seems that it it registering all the properties except `resourceGroupName`.

Thanks

**Comment by stack72** [comment]
Hi @yellowhat 

Please can you give us a sample of code that you are seeing this with so we can try and recreate it as well as what version of Pulumi and pulumi-azure that you are using

thanks

Paul

**Comment by yellowhat** [comment]
Dear @stack72,
I have tried to deploy just an `ActionGroup` but it worked without any problem:
```python
import pulumi_azure as azure

# Resource Group
anf_rg = azure.core.ResourceGroup(
    "rg",
    name="zz",
    location="westeurope",
)

anf_action_group = azure.monitoring.ActionGroup(
    "anf_action_group",
    name="action_mail",
    short_name="action_mail",
     resource_group_name=anf_rg.name,
    email_receivers=[
       azure.monitoring.ActionGroupEmailReceiverArgs(
            name="send2me",
            email_address="me@email.com",
        ),
    ],
)
```

Instead if I deploy it as part of my bigger deployment I get error.
The other interesting thing is that the exacly same code was working some days ago, not sure which pulumi* version I was using.

**Comment by yellowhat** [comment]
Forgot to add, I am using:
- pulumi: 3.2.1
- pulumi-azure: 4.1.0

**Comment by yellowhat** [comment]
Fixed in 4.2.0.

Thanks
